app/models/account.rb

```diff
@@ -28,6 +28,7 @@ class Account < ApplicationRecord
   has_many :stages, dependent: :destroy
   has_many :products, dependent: :destroy
   has_many :embedding_documments, dependent: :destroy
+  has_many :deal_products
```

Comment:
```suggestion
  has_many :deal_products, dependent: :destroy
```

Para quando deletarmos a conta deletar esses relacionamentos também